### PR TITLE
Up psalm to level 2

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because it should be bc.

## Changelog

```markdown
### Fixed
- Deprecations from Doctrine DBAL and ORM.
```

### CODE REVIEW

File: src/Collection/AuditedCollection.php [hunk 1]
```diff
@@ -346,14 +346,16 @@ public function partition(\Closure $p)
...
-    public function indexOf($element)
+    public function indexOf(mixed $element)
     {
         $this->forceLoad();
 
         return $this->loadedEntities->indexOf($element);
     }
 
     /**
+     * @param int $offset
```
Comment (jordisala1991) [inline src/Collection/AuditedCollection.php]:
```suggestion
     * @param int $offset
     * @param int|null $length
```

File: src/EventListener/LogRevisionsListener.php [hunk 2]
```diff
@@ -612,19 +613,20 @@ private function saveRevisionEntityData(EntityManagerInterface $em, ClassMetadat
...
-    private function recordRevisionForManyToManyEntity(object $relatedEntity, Connection $conn, string $revType, array $entityData, array $assoc, ClassMetadata $class, ClassMetadata $targetClass): void
+    private function recordRevisionForManyToManyEntity(object $relatedEntity, EntityManagerInterface $em, string $revType, array $entityData, array $assoc, ClassMetadata $class, ClassMetadata $targetClass): void
     {
+        $conn = $em->getConnection();
         $joinTableParams = [$this->getRevisionId($conn), $revType];
         $joinTableTypes = [\PDO::PARAM_INT, \PDO::PARAM_STR];
         foreach ($assoc['relationToSourceKeyColumns'] as $targetColumn) {
             $joinTableParams[] = $entityData[$class->fieldNames[$targetColumn]];
-            $joinTableTypes[] = $class->getTypeOfColumn($targetColumn);
+            $joinTableTypes[] = PersisterHelper::getTypeOfColumn($targetColumn, $targetClass, $em);
```
Comment (jordisala1991) [inline src/EventListener/LogRevisionsListener.php]:
```suggestion
            $joinTableTypes[] = PersisterHelper::getTypeOfColumn($targetColumn, $class, $em);
```

**Comment by jordisala1991** [comment]
This PR will need a good squash merge when its ready.

File: rector.php [hunk 3]
```diff
@@ -39,7 +39,9 @@
         CountOnNullRector::class,
         ExceptionHandlerTypehintRector::class,
         UnionTypesRector::class => [
+            __DIR__.'/src/AuditConfigurator.php',
```
Comment (jordisala1991) [inline rector.php]:
```suggestion
            __DIR__.'/src/AuditConfiguration.php',
```

**Review by VincentLanglet — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Pull request merged**